buddhi/lib/amp/toolkit/buddhi/service.rb

```diff
@@ -36,9 +36,29 @@ def cleaned_pattern(pattern)
         end
 
         def mapping_rules
+          mapping_rule_list = product_mapping_rules + backend_mapping_rules
+        end
+
+        def product_mapping_rules
           client.list_mapping_rules service_id
         end
 
+        def backend_mapping_rules
+          backend_usages.flat_map do |backend_usage|
+            client.list_backend_mapping_rules(backend_usage.fetch('backend_id')).map do |mp_rule|
+              u = URI.join('http://exemple.com', backend_usage.fetch('path'), mp_rule['pattern'])
```

Comment:
Why do we modify the returned backend mapping rule pattern with the contents of the backend usage and the own mapping rule original value?

Reply:
backend has a `base_path` that has to be prefixed to all mapping rules of the backend.

Reply:
I understand you are referring to the backend usage.
If I understand the model correctly, backend usage is not a property of a Backend but of a Product that is associated with a backend. It is how a request to a product is directed to a specific backend associated to the product.

Shouldn't Backend mapping rules be independent of the product they belong?

Reply:
Backend mapping rules are  independent of the product. 

`Service.backend_mapping_rules` are all the backend mapping rules a given product can access.

product -> backend_usages -> backend

Reply:
Ok, I understand then that they are the backend mapping rules in the context of a product, not the backend mapping rules of the backend per se, as they don't contain any product-specific path part :+1:

Reply:
`backend_usages` are specific for that product.